src/components/TransactionHistoryRows/TransactionDetails/ApprovalAmount.tsx

```diff
@@ -0,0 +1,63 @@
+import { MaxUint256 } from '@ethersproject/constants'
+import { AssetId } from '@shapeshiftoss/caip'
+import { TxMetadata } from '@shapeshiftoss/chain-adapters'
+import { useMemo } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { bn, bnOrZero } from 'lib/bignumber/bignumber'
+import { selectAssetById, selectMarketDataById } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { Row } from './Row'
+import { Text } from './Text'
+
+export const ApprovalAmount = ({
+  assetId,
+  value,
+  isRevoke,
+  parser,
+}: {
+  assetId: AssetId
+  value: string
+  isRevoke: boolean
+  parser: TxMetadata['parser'] | undefined
+}) => {
+  const translate = useTranslate()
+
+  const approvedAsset = useAppSelector(state => selectAssetById(state, assetId))
+  const approvedAmount = bn(value).div(bn(10).pow(approvedAsset.precision)).toString()
+  const approvedAssetMarketData = useAppSelector(state => selectMarketDataById(state, assetId))
+
+  const amountOrDefault = useMemo(() => {
```

Comment:
We don't do this currently, but I'll throw it out there:

Best practice for us in these cases is likely to make this a pure function (in this case with args `approvedAssetMarketData`, `approvedAmount`, and `approvedAsset`) that has unit tests so we can prove this does what it says it does.

It's a tree in a forest at this point, but something to consider.

Reply:
Very good idea ser.
I had concerns about passing `isRevoke` down which seems weird given the fact we're doing approval amount checks here already so we might as well check for this one, but at the same time we already had it in the parent so it made sense to pass it down to avoid duplication.
Making this the call of a pure, testable function seems to indicate it should not be passed down but rather re-defined|checked here! I'll get that spinned up shortly 🙏🏽

Reply:
Refactored and added tests in 456e6784ae02d036b4c41667f3c3ec3f91a3024a - no more passing `isRevoke` down!

Reply:
That is exactly what I had in mind, what an excellent commit!